Add why auth is needed for maps and search functionality (#3834)
aws-amplify/docs: src/pages/cli/geo/search.mdx

> Location search (search by places, addresses, coordinates)
```
If you haven't set up the `auth` category already, the Amplify CLI will guide you to enable the auth category.

> We would need the `auth` category added to your application so that appropriate permissions to search for places can be given to Authorized and/or Guest users as described below.

Reviewer: ```suggestion
The `auth` category is required in your application so that the appropriate permissions to render the map can be given to Authorized and/or Guest users as described below.
```
